AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/AudioPlayerTest.java

```diff
@@ -0,0 +1,146 @@
+package com.ichi2.anki.multimediacard;
+
+import com.ichi2.anki.RobolectricTest;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import org.robolectric.shadows.ShadowMediaPlayer;
+import org.robolectric.shadows.util.DataSource;
+
+import java.io.File;
+import java.io.IOException;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import static com.ichi2.testutils.AnkiAssert.assertDoesNotThrow;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class AudioPlayerTest extends RobolectricTest {
+
+    private AudioPlayer mAudioPlayer;
+    private File mFile;
+
+    @Before
+    public void before() throws IOException {
+        mAudioPlayer = new AudioPlayer();
+        mFile = tempFolder.newFile("testaudio.wav");
+
+        ShadowMediaPlayer testPlayer = new ShadowMediaPlayer();
+        DataSource mFileSource = DataSource.toDataSource(mFile.getAbsolutePath());
+        ShadowMediaPlayer.MediaInfo mFileInfo = new ShadowMediaPlayer.MediaInfo();
+        testPlayer.addMediaInfo(mFileSource, mFileInfo);
+    }
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
```

Comment:
We generally put class members on top of methods. Just a nitpick

```suggestion
    @Rule
    public TemporaryFolder tempFolder = new TemporaryFolder();

    @Before
    public void before() throws IOException {
        mAudioPlayer = new AudioPlayer();
        mFile = tempFolder.newFile("testaudio.wav");

        ShadowMediaPlayer testPlayer = new ShadowMediaPlayer();
        DataSource mFileSource = DataSource.toDataSource(mFile.getAbsolutePath());
        ShadowMediaPlayer.MediaInfo mFileInfo = new ShadowMediaPlayer.MediaInfo();
        testPlayer.addMediaInfo(mFileSource, mFileInfo);
    }

```